### ISSUE TITLE
the InsertWithOutput[T] method generates wrong sql insert statement.

### ISSUE THREAD

**Issue by msavencov** [linq2db/linq2db#3581]
Hello, 

I have a model with complex property like this one: 

```cs
    public class TestTable
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        
        public ExternalId ExternalId { get; set; } = new();
    }
    
    public class ExternalId
    {
        public string Id { get; set; }
        public string Source { get; set; } 
    }
```

and mapping: 
```cs
public class BaseConnection : DataConnection
{
    public BaseConnection(string providerName, string connectionString, MappingSchema mappingSchema) 
        : base(providerName, connectionString, mappingSchema)
    {
    }

    protected static readonly Lazy<MappingSchema> Schema = new Lazy<MappingSchema>(SchemaFactory);
    
    protected static MappingSchema SchemaFactory()
    {
        var schema = new MappingSchema();
        var builder = schema.GetFluentMappingBuilder();

        builder.Entity<TestTable>()
               .HasTableName("TestTable")
               .HasPrimaryKey(t => t.Id)
               .Property(t => t.Name).HasLength(100)
               .Property(t => t.ExternalId.Id).HasColumnName("ExternalId").HasLength(50)
               .Property(t => t.ExternalId.Source).HasColumnName("ExternalSource").HasLength(50)
            ;

        return builder.MappingSchema;
    }
}        
```
and context:
```cs
public class SqlServerConnection : BaseConnection
{
    public SqlServerConnection(string connectionString) 
        : base(ProviderName.SqlServer2019, connectionString, Schema.Value)
    {
            
    }
}
```

I'm trying to insert an entity with output using `LinqExtensions.InsertWithOutput<TTarget>`  method.
```cs
var connection = new SqlServerConnection("Server=localhost;Database=test;Integrated Security=True;");
var row = new TestTable {Id = Guid.NewGuid(), Name = "John Doe", ExternalId = {Id = "1", Source = "unknown"}};
var created = connection.GetTable<TestTable>().InsertWithOutput(row);
```

Both providers i tried, generates the instert query without `ExternalId` and `ExternalSource` columns.
But any other methods (like `IQueriable.ToList[T]`, `DataExtensions.Insert[T]` ) works as expected.

The generated sql instert statement by `.InsertWithOutput(row)` using `ProviderName.SqlServer2019`
```sql
// Sql Server
INSERT INTO [TestTable]
(
	[Id],
	[Name]
)
OUTPUT
	[INSERTED].[Id],
	[INSERTED].[Name]
VALUES
(
	@Id,
	@Name
)
```

The generated sql instert statement by `.InsertWithOutput(row)` using `ProviderName.SQLite`
```sql
INSERT INTO [TestTable]
(
	[Id],
	[Name]
)
VALUES
(
	@Id,
	@Name
)
RETURNING
	[TestTable].[Id],
	[TestTable].[Name],
	[TestTable].[ExternalId],
	[TestTable].[ExternalSource]
``` 

The `InsertWithOutput`  method throws an exception:
```
Exception message: Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'ExternalId', table 'test.dbo.TestTable'; column does not allow nulls. INSERT fails.
Stack trace:
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStat
eObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryHasMoreRows(Boolean& moreRows)
   at Microsoft.Data.SqlClient.SqlDataReader.TryReadInternal(Boolean setTimeout, Boolean& more)
   at Microsoft.Data.SqlClient.SqlDataReader.Read()
   at LinqToDB.Linq.QueryRunner.ExecuteQuery[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumbe
r)+MoveNext()
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found)
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at LinqToDB.LinqExtensions.InsertWithOutput[TTarget](ITable`1 target, TTarget obj)
   at ConsoleApp4.Program.Main(String[] args) in W:\github.com\msavencov\linq2db-sqlite-repro\ConsoleApp4\Program.cs:line 33
```

### Steps to reproduce
I've created a dummy project [msavencov/linq2db-sqlite-repro](https://github.com/msavencov/linq2db-sqlite-repro/tree/master/ConsoleApp4) that reproduces the bug (hope this is not a bug :)

```
git clone https://github.com/msavencov/linq2db-sqlite-repro.git
cd linq2db-sqlite-repro\ConsoleApp4
dotnet run
```

### Environment details
linq2db version: 3.7.0, 4.0.0
Database Server: Sql Server 2019, SQLite
Database Provider: System.Data.SQLite 1.0.115.5
Database Provider: System.Data.SQLite.Core 1.0.115.5
Operating system: Windows 10
.NET Framework: NET5.0, NET6.0

**Comment by shanmaa** [comment]
Hi,
May I know if this issue is resolved or are there any alternative to send a new record to insert method without creating a new instance.

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`
